enhancements/local-storage/health-monitoring.md

```diff
@@ -0,0 +1,132 @@
+- As an admin, I would like if components (such as OpenShift Data Foundation) could access storage device health and failover faster.
+
+
+### Implementation Details/Notes/Constraints
+
+`LocalVolumeDisocveryResults` already enumerate and refresh information about the disk in a per-node `LocalVolumeDiscoveryResult`, and a [PR](https://github.com/openshift/local-storage-operator/pull/249) is in progress to add metrics to each of the daemonsets that update the CR.
+
+`LocalVolumeDiscovery` can be configured via `nodeSelector` and `toleration` to match any set of nodes or all nodes. We could have the metrics be for all nodes or just for the ones that LocalVolumeDiscovery is configured to watch. Whether this should be controlled by the same (singleton) CR is an [open question](#Open-Questions    )
+
+We will be collecting metrics via [libstoragemgmt](https://github.com/libstorage/libstoragemgmt) and calling it via [libstoragemgmt-golang](https://github.com/libstorage/libstoragemgmt-golang/).
+If a block device is a partion or logical volume, we will only expose health for the parent device.
+
+
+### Risks and Mitigations
+
+- There is some concern that this is out of scope for Local Storage Operator
+    - We are discussing [alternatives](#Alternatives).
+- There is some uncertainty about whether this implementation precludes other components correlating PersistentVolumes with this information
+    - Mitigation: We are investigating if it is an acceptable pattern for internal components to scrape and correlate cluster prometheus metrics for automation. There has been some indication that this is the case on #forum-monitoring
+- The golang bindings for libstoragemgmgt also needs some review and polish.
+    - Mitigation: The ODF(OpenShift Data Foundation) team will contribute to and help the RHEL team maintain libstoragemgmt golang
+
+
+## Design Details
+
+### Open Questions
+
+- Is Local Storage Operator the best place for this?
+  - see [alternatives](#Alternatives)
+- Should we allow configuration to watch only a subset of nodes or watch all nodes by default? Should the metrics be controller by LocalVolumeDiscovery (singleton)
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+- E2E tests will deploy the component and verify that metrics are being emitted per device
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+
+
+### Upgrade / Downgrade Strategy
+
+
+
+### Version Skew Strategy
+
+
+
+## Implementation History
+
+
+## Drawbacks
+
+
+
+## Alternatives
+
+- We explored node_exporter, but it is a rootless container, so SMART data can't be checked.
+- We considered node-problem-detector, but it doesn't export metrics only sets node conditions and creates events.
+- We *could* create a new component for this.
```

Comment:
I'm missing some reasoning of why LSO is _not_ the right place.

While I agree that other components can own disks of a host, LSO is one of them, thus it looks like to be like a good starting point at least. The good is alos that LSO can correlate it to PVs

Reply:
In general argument against including these metrics in LSO is - LSO is not a general purpose disk management solution. It exists to allow creation of PVs from user selected disks. In other words - it only concerns itself with PV creation from selected disks and if task was to monitor health of PVs that LSO creates, it would be a suitable place to add health checks. 

In this case though - task is to monitor health of all disks, including root disk, boot disk etc and in which case - LSO does not seem like a suitable place to report such data.  There are some advantages of keeping LSO lightweight and easier to maintain and using it as a kitchen sink makes maintenance difficult.

Reply:
@gnufied I'm not sure I understand the consequences, the solution has to be maintained somewhere. Can you clarify the possible negative situations from putting it in LSO so I can put it in the enhancement?

Reply:
If LSO creates PVs from local disks, I think it seems to be the place to monitor health of those disks as well. Like Rohan, I haven't quite understood the argument against adding this feature to LSO.